pkg/apis/extensions/v1beta1/types.go

```diff
@@ -1102,3 +1103,78 @@ type PodSecurityPolicyList struct {
 	// Items is a list of schema objects.
 	Items []PodSecurityPolicy `json:"items" protobuf:"bytes,2,rep,name=items"`
 }
+
+// +genclient=true
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	v1.ObjectMeta        `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+
+	Spec TemplateSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
```

Comment:
These comments appear in user-facing documentation. Please use json field names and plain English to describe fields and types, or don't mention the field name at all. 

Also, we usually don't specify optional vs. required in the comment.That's indicated by the presence or absence of `omitempty`.